### ISSUE TITLE
Rewrite animated portrait system.

### ISSUE THREAD

**Issue by Windblade-GR01** [Windblade-GR01/Ikemen-GO#14]
In the S-SiZe Ikemen system here is a batch process that uses Mugen's sprites tools to export all sprites from a char sff, put some specific sprites into a new sff, and then lua loads those new sff at the beginning to show animated portraits.

There problems with this and is that:
1. We depend of Elecbyte proprietary tools.
2. Is limited to windows only.

A rewrite with a better implementation should be considered.

**Comment by cyliamargatroid** [comment]
My Idea is to Make it Manually instead of Batch, Kinda Similar to ShugenDO and Tutorials for MugenHook Select Screen
In kfm.def
```
; Definition file for Kung Fu Man
; Contains all the filenames needed for the character

; Player information
[Info]
name = "Kung Fu Man"        ;Name of character
displayname = "Kung Fu Man" ;Name of character to display
versiondate = 04,14,2002    ;Version date of character (MM-DD-YYYY)
mugenversion = 04,14,2002   ;Version of M.U.G.E.N character works on
author = "Elecbyte"         ;Character author name
pal.defaults = 6,3,4,2      ;Default palettes in order of preference

; Files for the player
[Files]
cmd     = kfm.cmd   ;Command set
cns     = kfm.cns   ;Constants
st      = kfm.cns   ;States
stcommon = common1.cns ;Common states
sprite  = kfm.sff   ;Sprite
anim    = kfm.air   ;Animation
sound   = kfm.snd   ;Sound
pal1    = kfm6.act  ;Palettes (can have up to 12)
pal2    = kfm4.act
pal3    = kfm2.act
pal4    = kfm5.act
pal5    = kfm3.act
pal6    = kfm.act

; Arcade mode
[Arcade]
intro.storyboard = intro.def
ending.storyboard = ending.def

[animatedselect]
sprite = show_menu.ssf
anim = show_menu.air
; SelectOrder: Idle, Selected, Victory, Continue, Game over
selectorder = 0,1,0,5500,5500
```
in show_menu.air
```
; Character Pose
[Begin Action 0 ]
0, 0, 0 ,0, 10
0, 1, 0 ,0, 7
0, 2, 0 ,0, 7
0, 3, 0 ,0, 7
0, 4, 0 ,0, 7
0, 5, 0 ,0, 45
0, 4, 0 ,0, 7
0, 3, 0 ,0, 7
0, 2, 0 ,0, 7
0, 1, 0 ,0, 7
0, 0, 0 ,0, 40

; Selected
[Begin Action 1]
200, 0, 0 ,0, 4
181, 0, 0 ,0, 5
181, 1, 0 ,0, 5
181, 2, 0 ,0, 5
181, 3, 0 ,0, 5
181, 4, 0 ,0, 5
181, 5, 0 ,0, 5
181, 6, 0 ,0, 30
181, 5, 0 ,0, 7
181, 4, 0 ,0, 6
181, 3, 0 ,0, 5
181, 2, 0 ,0, -1
181, 2, 0 ,0, -1


; Continue Screen
[Begin Action 5300]
5300,0, 0,0, 8
5300,1, 0,0, 8
5300,2, 0,0, 8
5300,3, 0,0, 8
5300,2, 0,0, 8
5300,1, 0,0, 8
```

**Comment by K4thos** [comment]
animated portraits preloading and character snd file being used upon selection are already available since version 0.96.
